helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java

```diff
@@ -300,6 +300,9 @@ public void addJob(final Job job) throws JobExistsException {
         }
         throw new JobExistsException(id.toString());
       }
+    } catch (NoNodeException e) {
```

Comment:
It's the `client.transaction()` that throws the `NoNodeException` right? How can this happen with `ensurePath()`?